lib/resque/plugins/retry.rb

```diff
@@ -274,11 +274,9 @@ def retry_criteria_valid?(exception, *args)
         retry_based_on_criteria = false
         unless retry_based_on_exception
           # call user retry criteria check blocks.
-          retry_criteria_checks.each do |criteria_check|
-            retry_based_on_criteria ||= !!instance_exec(exception, *args, &criteria_check)
-          end
+          retry_based_on_criteria = retry_criteria_checks_pass?(exception, *args)
+          log_message "user retry criteria is #{retry_based_on_criteria ? '' : 'not '}sufficient for a retry", args, exception
```

Comment:
I moved the log message inside; because I thought it might be misleading. Previously, if `retry_based_on_exception = true`, then this log message would *always* say that the retry criteria is *not* sufficient for a retry, even though it technically might have been.